### PR TITLE
Fix redirects for gem browser and map viewer

### LINKED ISSUE #1292: broken link missing redirect
**Describe the bug**

Links to the Map Viewer containing the `compartment` or `subsystem` keywords are broken, e.g. [https://www.metabolicatlas.org/explore/map-viewer/human1/compartment/endoplasmic_reticulum?dim=2d](https://www.metabolicatlas.org/explore/map-viewer/human1/compartment/endoplasmic_reticulum?dim=2d).

**Expected behavior**

Redirects in `router.js` for:
- [ ] all Map Viewer links that follow the old style of including `compartments` and `subsystems` to be working

**Additional context**

It seems the existing rule is not entirely correct anymore: https://github.com/MetabolicAtlas/MetabolicAtlas/blob/f37d488a3899127beaf8537beb278095f1b28b75/frontend/src/router.js#L106-L109

### PR DESCRIPTION
**Pull request by jonandernovella**
**Related issue(s) and PR(s)** 

This PR closes #1292.

**Type of change**  

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other
 
**List of changes made**  

- Updated redirects for map-viewer so that users end up at right URL without component and subsystem keywords
- Update redirects for map-viewer and genome-browser so that users are redirected to the respective resources rather than ending up in an endless loading page

In short:

- Check for when only gem browser or map viewer is visited
- Make sure the compartment and subsystem keywords are dropped

**Testing**  

Navigate to:

- [/explore/map-viewer/human1/compartment/endoplasmic_reticulum?dim=2d](https://www.metabolicatlas.org/explore/map-viewer/human1/compartment/endoplasmic_reticulum?dim=2d).
- [/explore/map-viewer/human1/)
- [/explore/gem-browser/human1/)

### CODE REVIEW

**Review by perjo — APPROVED** [review]
Looks good and seems to fix problem. Still feel a little bit of FUD so would welcome more reviews.

**Review by kusalananda — APPROVED** [review]
Good after demoing the fix.
